pkg/api/types.go

```diff
@@ -450,6 +452,16 @@ type GitRepoVolumeSource struct {
 type SecretVolumeSource struct {
 	// Name of the secret in the pod's namespace to use
 	SecretName string `json:"secretName"`
+	// Modes describes access permissions to be applied per Secret's data key while mounting
+	Modes []SecretFileMode `json:"modes"`
```

Comment:
If we provide users with option to put multiple data we should allow them to define different permissions as well. On the other hand I can think of that as a grouping factor, where you'd group kind of "similar" data into one secret. Before doing any changes I'll wait for the outcome from #7925 @erictune mentioned.